docs/dev/guides/fragment-controllers.md

```diff
@@ -0,0 +1,162 @@
+---
+title: "Fragment Controllers"
+description: "The concept behind fragment controllers."
+aliases:
+  - /guides/fragment-controllers
+---
+
+
+## Intro
+
+A regular page in Contao consists of various partials that build the actual page content:
+
+1. The page has a layout assigned, which renders the base template (usually `fe_page.html5`)
+2. Each page layout has multiple layout sections (header, left, main, custom ones etc.)
+3. Each layout sections contain one or multiple modules or articles
+4. Each article is a container filled with content elements
+
+Splitting the page into these partials is the foundation of flexible page building in Contao.
+
+{{% notice note %}}
+The concept of modules and elements exists since the beginning of Contao.
+Front end modules and content elements as Symfony controller fragments are available since **Contao 4.5**.
+{{% /notice %}}
+
+{{% notice warning %}}
+Before you learn about fragments, make sure to
+[understand Symfony controllers](https://symfony.com/doc/current/controller.html).
+{{% /notice %}}
+
+
+## Concept
+
+In Symfony, an HTML template is usually generated by a controller. From the Symfony documentation:
+
+> A controller is a PHP function you create that reads information from the Request object and
+> creates and returns a Response object. The response could be an HTML page, JSON, XML, a
+> file download, a redirect, a 404 error or anything else. The controller executes whatever
+> arbitrary logic your application needs to render the content of a page.
+
+A normal controller has a route – a URL on your site – that will generate the response.
+As fragments are partials of an existing page; they do not have a route, but same as a
+normal controller they will get a request and generate a response. This concept is called
+[*Sub Requests* in Symfony][subrequests], each fragment in Contao is a sub request in Symfony.
+
+From an HTTP perspective, each (sub) request is totally independent. Each one will get request
+parameters (like `GET` and `POST` data), and they can return HTTP headers (like `Content Type`
+or `Cache-Control`).
+
+{{% notice note %}}
+Since **Contao 4.9**, subrequests can generate [a response that modifies the cache time of
+the main response](/framework/caching/#caching-fragments).
+{{% /notice %}}
+
+
+
+## Fragment Controllers
+
+Contao provides two different types of fragments out-of-the-box:
+[Front End Modules][moduels] and [Content Elements][elements]. For each of
+these, there are abstract classes (namely `AbstractFrontendModuleController`
+and `AbstractContentElementController`) that mimick the behavior of a Contao 3
+module/element. They generate the Contao template based on the request and current
```

Comment:
I don’t think mentioning v3 is relevant

Reply:
what else would you call it? *old-school Contao module/element*? Or legacy?

Reply:
nevermind, updated in 0f29eab5055d1f6ea24c00ac6ab144527302da84 :)